tests/external_executor/executors/util.py

```diff
@@ -33,8 +53,78 @@ def terminate(self):
 
 
 @contextmanager
-def executor_thread(executor):
+def executor_thread(executor: Executor):
     et = ExecutorThread(executor)
     et.start()
     yield executor
     et.terminate()
+
+
+class ExecutorContainer:
+    def __init__(
+        self,
+        executor: str,
+        node_public_rpc_address: str,
+        network: Network,
+        supported_endpoints: Set[Tuple[str, str]],
+    ):
+        self._client = docker.DockerClient()
+
+        image_name = "ccfmsrc.azurecr.io/ccf/ci:14-02-2023-virtual"
```

Comment:
I think this could/should be a smaller image to mimic a real scenario, e.g. `mcr.microsoft.com/cbl-mariner/base/python:3`?

Reply:
Will do in a follow up PR